### ISSUE TITLE
Filter on "Storage filename"

### ISSUE THREAD

**Issue by SheldonWBM** [heartexlabs/label-studio#4824]
**Describe the bug**
I am unable to filter on the "storage filename" using Amazon S3 for source storage.
If it is not implemented, it should be removed as an option from the filter list.
With over 200,000 images, it is the only way to filter/sort images as the "storage filename" retains the original directory structure.  

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to Filters
2. Click on 'Where Storage filename [any option] '

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/sql/query.py", line 1955, in add_fields
    join_info = self.setup_joins(name.split(LOOKUP_SEP), opts, alias, allow_many=allow_m2m)
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/sql/query.py", line 1648, in setup_joins
    path, final_field, targets, rest = self.names_to_path(
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/sql/query.py", line 1562, in names_to_path
    raise FieldError("Cannot resolve keyword '%s' into field. "
django.core.exceptions.FieldError: Cannot resolve keyword 'storage_filename' into field. Choices are: annotations, cancelled_annotations, comment_authors, comment_count, created_at, data, drafts, file_upload, file_upload_id, id, inner_id, io_storages_azureblobimportstoragelink, io_storages_gcsimportstoragelink, io_storages_localfilesimportstoragelink, io_storages_redisimportstoragelink, io_storages_s3importstoragelink, is_labeled, last_comment_updated_at, locks, meta, overlap, predictions, project, project_id, total_annotations, total_predictions, unresolved_comment_count, updated_at, updated_by, updated_by_id

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/label-studio/label_studio/data_manager/api.py", line 210, in get
    queryset = self.get_task_queryset(request, prepare_params)
  File "/label-studio/label_studio/data_manager/api.py", line 178, in get_task_queryset
    return Task.prepared.only_filtered(prepare_params=prepare_params)
  File "/label-studio/label_studio/data_manager/managers.py", line 660, in only_filtered
    return queryset.prepared(prepare_params=prepare_params)
  File "/label-studio/label_studio/data_manager/managers.py", line 471, in prepared
    queryset = apply_filters(queryset, prepare_params.filters, project, request)
  File "/label-studio/label_studio/data_manager/managers.py", line 349, in apply_filters
    value_type = type(queryset.values_list(field_name, flat=True)[0]).__name__
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/query.py", line 867, in values_list
    clone = self._values(*_fields, **expressions)
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/query.py", line 835, in _values
    clone.query.set_values(fields)
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/sql/query.py", line 2272, in set_values
    self.add_fields(field_names, True)
  File "/usr/local/lib/python3.10/dist-packages/django/db/models/sql/query.py", line 1982, in add_fields
    raise FieldError("Cannot resolve keyword %r into field. "
django.core.exceptions.FieldError: Cannot resolve keyword 'storage_filename' into field. Choices are: annotations, cancelled_annotations, comment_authors, comment_count, created_at, data, drafts, file_upload, file_upload_id, id, inner_id, io_storages_azureblobimportstoragelink, io_storages_gcsimportstoragelink, io_storages_localfilesimportstoragelink, io_storages_redisimportstoragelink, io_storages_s3importstoragelink, is_labeled, last_comment_updated_at, locks, meta, overlap, predictions, project, project_id, total_annotations, total_predictions, unresolved_comment_count, updated_at, updated_by, updated_by_id
```

**Screenshots**
![image](https://github.com/HumanSignal/label-studio/assets/127118806/9d630cb6-af44-4abc-b15c-8e934a91fd9b)

**Environment (please complete the following information):**
 - Docker: `heartexlabs/label-studio:develop` or  `heartexlabs/label-studio:latest`
 - Label Studio Version 1.9.1dev

**Additional context**
Likely related to the closed PR: #3494

**Comment by hogepodge** [comment]
Thanks for the bug report @SheldonWBM. We've verified this and have added it to our tracker to fix in an upcoming release.

**Comment by hogepodge** [comment]
Fixed upstream, this should be available in an upcoming Label Studio release (likely 1.10.3+, since this patch was added after the 1.10.2 cut)

**Comment by SheldonWBM** [comment]
Thank you and your team for releasing a patch @hogepodge  :+1:

**Comment by jombooth** [comment]
Hi @SheldonWBM - this was fixed in our latest release, [1.11.0](https://github.com/HumanSignal/label-studio/releases/tag/1.11.0)!

**Comment by SheldonWBM** [comment]
Hi @jombooth Thank you, we switched to 1.11.0  :+1:
